streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -332,20 +362,31 @@ private void connectProcessorAndStateStore(String processorName, String stateSto
      *
      * @return groups of topic names
      */
-    public Map<Integer, Set<String>> topicGroups() {
-        Map<Integer, Set<String>> topicGroups = new HashMap<>();
+    public Map<Integer, TopicsInfo> topicGroups() {
+        Map<Integer, TopicsInfo> topicGroups = new HashMap<>();
 
         if (nodeGroups == null)
             nodeGroups = makeNodeGroups();
 
         for (Map.Entry<Integer, Set<String>> entry : nodeGroups.entrySet()) {
-            Set<String> topicGroup = new HashSet<>();
+            Set<String> sourceTopics = new HashSet<>();
+            Set<String> stateTopics = new HashSet<>();
```

Comment:
This set contains state names, not topic names. The topic name is no longer same as the state name.